modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/MappingFeatureCollection.java

```diff
@@ -314,9 +314,12 @@ public boolean retainAll(Collection<?> c) {
      * @see org.geotools.feature.FeatureCollection#size()
      */
     public int size() {
-        // VT: The only way to count the size of the feature is by building it and that becomes very
-        // inefficient.
-        return 0;
+        try {
+            if (!store.isAllowCounts()) return 0;
+            return store.getCount(query);
+        } catch (IOException e) {
+            return 0;
```

Comment:
Log the exception.  In case this exception is a cancel trigger (I think it is since it should be a connection problem with database) we should re-throw the exception as a RuntimeException.

Reply:
fixed